code/modules/projectiles/updated_projectiles/gun_system.dm

```diff
@@ -423,14 +423,12 @@ User can be passed as null, (a gun reloading itself for instance), so we need to
 		"<span class='notice'>You cock [src], clearing a [in_chamber.name] from its chamber.</span>", null, 4)
 
 		// Get gun information from the current mag if its equipped otherwise the default ammo & caliber
-		var/bullet_ammo_type
+		var/bullet_ammo_type = in_chamber.ammo.type
 		var/bullet_caliber
```

Comment:
Alternative syntax, which is nice to know:
```suggestion
		var/bullet_caliber = current_mag ? current_mag.caliber : caliber
```

Reply:
I feel the comments are important, but thank you anyways.